MongoDB extension test failure in CI

**Describe the bug**
When working on https://github.com/quarkusio/quarkus/pull/3208, I got CI failure for Linux JDK build for MongoDB extension. 

**Expected behavior**
The CI build to succeed.

**Actual behavior**
The build fails: https://dev.azure.com/quarkus-ci/quarkus/_build/results?buildId=4430&view=logs.

**Environment (please complete the following information):**
 - ci (Linux jdk build) 

**Additional context**
Build failed with 
````
[mongod output] 2019-07-12T12:20:36.984+0000 I STORAGE  [conn28] WiredTigerKVEngine shutting down
[mongod output] 2019-07-12T12:20:37.064+0000 I STORAGE  [conn28] shutdown: removing fs lock...
[mongod output] 2019-07-12T12:20:37.064+0000 I CONTROL  [conn28] now exiting
[mongod output] 2019-07-12T12:20:37.064+0000 I CONTROL  [conn28] shutting down with code:0
[mongod output] 
[ERROR] Tests run: 13, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 4.91 s <<< FAILURE! - in io.quarkus.mongodb.BasicInteractionTest
[ERROR] testIndexCreation  Time elapsed: 0.454 s  <<< FAILURE!
java.lang.AssertionError: 

Expected size:<1> but was:<2> in:
<[{"key"={"_id"=1}, "name"="_id_", "ns"="mongo-extension-test-db.NPYADPQL", "v"=2},
    {"key"={"i"=1}, "name"="my-index", "ns"="mongo-extension-test-db.NPYADPQL", "v"=2}]>
  at io.quarkus.mongodb.BasicInteractionTest.testIndexCreation(BasicInteractionTest.java:310)
```

--- a/extensions/mongodb-client/runtime/src/test/java/io/quarkus/mongodb/BasicInteractionTest.java
+++ b/extensions/mongodb-client/runtime/src/test/java/io/quarkus/mongodb/BasicInteractionTest.java
@@ -306,7 +306,7 @@ class BasicInteractionTest extends MongoTestBase {
         assertThat(i).isEqualTo("my-index");
         assertThat(collection.listIndexes().toList().run().toCompletableFuture().join()).hasSize(2);
 
-        collection.dropIndex(i);
+        collection.dropIndex(i).toCompletableFuture().join();
         assertThat(collection.listIndexes().toList().run().toCompletableFuture().join()).hasSize(1);
 
     }

